src/main/java/redis/clients/jedis/Protocol.java

```diff
@@ -261,16 +261,20 @@ public static Object read(final RedisInputStream is) {
     GEORADIUSBYMEMBER, GEORADIUSBYMEMBER_RO, MODULE, BITFIELD, HSTRLEN, TOUCH, SWAPDB, MEMORY,
     XADD, XLEN, XDEL, XTRIM, XRANGE, XREVRANGE, XREAD, XACK, XGROUP, XREADGROUP, XPENDING, XCLAIM;
 
-    private final byte[] raw;
+    private byte[] raw;
 
     Command() {
-      raw = SafeEncoder.encode(this.name());
+      rename(this.name());
     }
 
     @Override
     public byte[] getRaw() {
       return raw;
     }
+
+    public void rename(final String newName) {
+      raw = SafeEncoder.encode(newName);
```

Comment:
Couldn't this be an issue in multithreading? Synchronized/Lock may be required here.

Reply:
You're right, but volatile should do, I'm trying to find a way to avoid the performance penalty on each getRaw() call

Reply:
* +1 to volatile (or AtomicReference), at the moment it's not though, so we need to update that to either volatile or AtomicReference.
* We should make this fail (and have tests for) if `newName` is:
  * null
  * empty
  * an existing command name